packages/near-api-js/src/utils/key_pair.ts

```diff
@@ -78,17 +108,20 @@ export abstract class KeyPair {
     static fromRandom(curve: string): KeyPair {
         switch (curve.toUpperCase()) {
             case 'ED25519': return KeyPairEd25519.fromRandom();
+            case 'SECP256K1': return KeyPairSecp256k1.fromRandom();
             default: throw new Error(`Unknown curve ${curve}`);
         }
     }
 
     static fromString(encodedKey: string): KeyPair {
         const parts = encodedKey.split(':');
+        //TODO: clarify what we must do here
```

Comment:
We shouldn't assume what the caller wants us to do so I think we should just remove this check and require a prefixed key be passed in, i.e. `secp256k1:` or `ed25519:`.

We already throw an error below for misformed keys, but we can go a step further and introduce a [Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) to enforce it at the type level:
```ts
static fromRandom(curve: `${'secp256k1' | 'ed25519'}:${stirng}`) {
  // ...
}

Reply:
Agreed 👍

Reply:
We can delete this comment now :)